File: Components/WinForms/VB/TreeListTutorials/DataMode/DataMode.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data

Namespace DevExpress.XtraTreeList.Demos.Tutorials
	''' <summary>
	''' Summary description for DataMode.
	''' </summary>
	Public Partial Class DataMode
		Inherits TutorialControl
		'<treeList1>
		Public Sub New()
			InitializeComponent()
			InitData()
		End Sub
		'</treeList1>
        Public Overrides ReadOnly Property SetNewWhatsThisPadding() As Boolean
            Get
                Return True
            End Get
        End Property
		'<treeList1>
		Private Sub InitData()
			Dim DBFileName As String = DevExpress.Utils.FilesHelper.FindingFileName(Application.StartupPath, "Data\Departments.xml")
			If DBFileName <> "" Then
				Dim dataSet As DataSet = New DataSet()
				dataSet.ReadXml(DBFileName)
				treeList1.DataSource = dataSet.Tables(0).DefaultView
				treeList1.PopulateColumns()
				treeList1.BestFitColumns()
				treeList1.ExpandAll()
			End If
		End Sub
		'</treeList1>

        Private Sub DataMode_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
            If treeList1.Columns("Budget") Is Nothing Then
                Return
            End If
            treeList1.Columns("Budget").Format.FormatString = "c"
            treeList1.BestFitColumns()
        End Sub
	End Class
End Namespace

File: Components/WinForms/CS/TreeListTutorials/DataMode/DataMode.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace DevExpress.XtraTreeList.Demos.Tutorials {
    /// <summary>
    /// Summary description for DataMode.
    /// </summary>
    public partial class DataMode : TutorialControl {
        //<treeList1>
        public DataMode() {
            InitializeComponent();
            InitData();
        }
        //</treeList1>
        public override bool SetNewWhatsThisPadding { get { return true; } }
        //<treeList1>
        private void InitData() {
            string DBFileName = DevExpress.Utils.FilesHelper.FindingFileName(Application.StartupPath, "Data\\Departments.xml");
            if(DBFileName != "") {
                DataSet dataSet = new DataSet();
                dataSet.ReadXml(DBFileName);
                treeList1.DataSource = dataSet.Tables[0].DefaultView;
                treeList1.PopulateColumns();
                treeList1.BestFitColumns();
                treeList1.ExpandAll();
            }
        }
        //</treeList1>

        private void DataMode_Load(object sender, System.EventArgs e) {
            if(treeList1.Columns["Budget"] == null) return;
            treeList1.Columns["Budget"].Format.FormatString = "c";
            treeList1.BestFitColumns();
        }
    }
}
